File: projects/hangman.js
const themes = {
  animals: ["alligator", "ant", "bat", "bear", "beaver", "cat", "dog", "eagle", "elephant", "giraffe", "lion", "monkey", "panda", "penguin", "rabbit", "tiger", "wolf", "zebra"],
  food: ["apple", "banana", "carrot", "cheese", "cookie", "fries", "mango", "milk", "onion", "pizza", "salad", "sushi"],
  ice_cream: ["vanilla", "chocolate", "strawberry", "mint", "rockyroad", "butterpecan"],
  instruments: ["guitar", "piano", "drums", "violin", "flute", "saxophone"]
};

const hangmanArt = [
`  +---+
  |   |
      |
      |
      |
      |
=========`,
`  +---+
  |   |
  O   |
      |
      |
      |
=========`,
`  +---+
  |   |
  O   |
  |   |
      |
      |
=========`,
`  +---+
  |   |
  O   |
 /|   |
      |
      |
=========`,
`  +---+
  |   |
  O   |
 /|\\  |
      |
      |
=========`,
`  +---+
  |   |
  O   |
 /|\\  |
 /    |
      |
=========`,
`  +---+
  |   |
  O   |
 /|\\  |
 / \\  |
      |
=========`
];

let answer = "";
let guessedLetters = [];
let wrongGuesses = [];

function startGame() {
  const theme = document.getElementById("theme-select").value;
  const words = themes[theme];
  answer = words[Math.floor(Math.random() * words.length)];
  guessedLetters = [];
  wrongGuesses = [];
  document.getElementById("message").textContent = "";
  document.getElementById("wrong-letters").textContent = "";
  updateWordDisplay();
  updateHangman();
  createLetterButtons();
}

function updateWordDisplay() {
  const display = answer
    .split("")
    .map(letter => guessedLetters.includes(letter) ? letter : "_")
    .join(" ");
  document.getElementById("word-display").textContent = display;

  if (!display.includes("_")) {
    document.getElementById("message").textContent = "🎉 You won!";
    disableAllButtons();
  }
}

function updateHangman() {
  const wrongCount = wrongGuesses.length;
  document.getElementById("hangman-art").textContent = hangmanArt[wrongCount];
  document.getElementById("wrong-letters").textContent = wrongGuesses.join(", ");
  if (wrongCount >= hangmanArt.length - 1) {
    document.getElementById("message").textContent = `💀 You lost! The word was: ${answer} 💀`;
    disableAllButtons();
  }
}

function createLetterButtons() {
  const container = document.getElementById("letters");
  container.innerHTML = "";
  for (let i = 65; i <= 90; i++) {
    const letter = String.fromCharCode(i).toLowerCase();
    const button = document.createElement("button");
    button.textContent = letter;
    button.onclick = () => handleGuess(letter, button);
    container.appendChild(button);
  }
}

function handleGuess(letter, button) {
  button.disabled = true;
  if (answer.includes(letter)) {
    guessedLetters.push(letter);
    updateWordDisplay();
  } else {
    if (!wrongGuesses.includes(letter)) {
      wrongGuesses.push(letter);
      updateHangman();
    }
  }
}

function disableAllButtons() {
  document.querySelectorAll("#letters button").forEach(button => {
    button.disabled = true;
  });
}

File: projects/hangman.py
import random
from colorama import Fore, Style, init

init(autoreset=True)

# Word themes (same as before)
themes = {
    "animals": ["alligator", "ant", "anteater", "antelope", "ape", "baboon", "badger", "bat", "bear", "beaver", "bee", "bison", "boar", "buffalo", "butterfly", "camel", "capybara", "caribou", "cat", "caterpillar", "cattle", "cheetah", "chicken", "chimpanzee", "clam", "cobra", "cockroach", "coyote", "crab", "crane", "crocodile", "crow", "deer", "dinosaur", "dog", "dolphin", "donkey", "dormouse", "dove", "dragonfly", "duck", "eagle", "echidna", "eel", "elephant", "elk", "emu", "falcon", "ferret", "finch", "fish", "flamingo", "fly", "fox", "frog", "gazelle", "gerbil", "giraffe", "goat", "goldfinch", "goldfish", "goose", "gorilla", "grasshopper", "gull", "hamster", "hare", "hawk", "hedgehog", "heron", "hippopotamus", "hornet", "horse", "human", "hummingbird", "hyena", "jackal", "jaguar", "jay", "jellyfish", "kangaroo", "koala", "lemur", "leopard", "lion", "llama", "lobster", "locust", "manatee", "mantis", "meerkat", "mole", "mongoose", "monkey", "moose", "mosquito", "mouse", "mule", "narwhal", "newt", "nightingale", "octopus", "opossum", "ostrich", "otter", "owl", "ox", "oyster", "panda", "panther", "parrot", "pelican", "penguin", "pig", "pigeon", "polar-bear", "pony", "porcupine", "porpoise", "quail", "quetzal", "rabbit", "raccoon", "ram", "rat", "raven", "red-panda", "reindeer", "rhinoceros", "salamander", "salmon", "sardine", "scorpion", "seahorse", "seal", "shark", "sheep", "skunk", "snail", "snake", "sparrow", "spider", "squid", "squirrel", "starling", "stingray", "stoat", "stork", "swan", "termite", "tiger", "toad", "turkey", "turtle", "viper", "vulture", "wallaby", "walrus", "wasp", "weasel", "whale", "wildcat", "wolf", "wolverine", "woodpecker", "worm", "yak", "zebra"],
    "food": ["apple", "banana", "pear", "mango", "orange", "carrot", "broccoli", "avocado", "tomato", "onion", "sourcream", "butter", "eggs", "yogurt", "milk", "cheese", "pizza", "sushi", "pancakes", "frenchtoast", "pasta", "burger", "salad", "fries", "cake", "cookie", "pie", "cupcake"],
    "ice_cream": ["vanilla", "chocolate", "strawberry", "blueberry", "cottoncandy", "mangonada", "mintchocolatechip", "mint", "cookiedough", "rockyroad", "cookiesandcream", "pistachio", "caramel", "fudge", "butterpecan"],
    "instruments": ["guitar", "piano", "drums", "violin", "viola", "triangle", "maracas", "clarinet", "flute", "trumpet", "saxophone", "cello", "harp", "bass"],
}

art = {
    0: (
        "  +---+",
        "  |   |",
        "      |",
        "      |",
        "      |",
        "      |",
        "========"
    ),
    1: (
        "  +---+",
        "  |   |",
        "  O   |",
        "      |",
        "      |",
        "      |",
        "========"
    ),
    2: (
        "  +---+",
        "  |   |",
        "  O   |",
        "  |   |",
        "      |",
        "      |",
        "========"
    ),
    3: (
        "  +---+",
        "  |   |",
        "  O   |",
        " /|   |",
        "      |",
        "      |",
        "========"
    ),
    4: (
        "  +---+",
        "  |   |",
        "  O   |",
        " /|\\  |",
        "      |",
        "      |",
        "========"
    ),
    5: (
        "  +---+",
        "  |   |",
        "  O   |",
        " /|\\  |",
        " /    |",
        "      |",
        "========"
    ),
    6: (
        "  +---+",
        "  |   |",
        "  O   |",
        " /|\\  |",
        " / \\  |",
        "      |",
        "========"
    )
}

def display_man(wrong_guesses):
    print(Fore.CYAN + "***********")
    for line in art[wrong_guesses]:
        print(Fore.RED + line + Style.RESET_ALL)
    print(Fore.CYAN + "***********")

def display_hint(hint):
    print(" ".join(hint))

def display_answer(answer):
    print("Answer was:", Fore.GREEN + " ".join(answer))

def choose_theme():
    print("Choose a word theme:")
    for i, theme in enumerate(themes.keys(), 1):
        print(f"{i}. {theme.capitalize()}")
    choice = int(input("Enter the number of your choice: "))
    theme_name = list(themes.keys())[choice - 1]
    return theme_name

# Main game function
def main():
    print("Hangman Game")
    while True:  # Loop to allow restarting the game
        theme_name = choose_theme()  # Player selects a theme
        words = themes[theme_name]  # Get the word list based on the selected theme

        answer = random.choice(words)
        hint = ["_"] * len(answer)
        wrong_guesses = 0
        guessed_letters = set()
        is_running = True

        while is_running:
            display_man(wrong_guesses)
            display_hint(hint)
            guess = input("Enter a letter: ").lower()

            if len(guess) != 1 or not guess.isalpha():
                print(Fore.YELLOW + "Invalid input. Please try again.")
                continue

            if guess in guessed_letters:
                print(Fore.YELLOW + f"{guess} has already been guessed. Enter a different letter.")
                continue

            guessed_letters.add(guess)

            if guess in answer:
                for i in range(len(answer)):
                    if answer[i] == guess:
                        hint[i] = guess
            else:
                wrong_guesses += 1

            if "_" not in hint:
                display_man(wrong_guesses)
                display_answer(answer)
                print(Fore.GREEN + "🎉 You won! 🎉")
                is_running = False
            elif wrong_guesses >= len(art) - 1:
                display_man(wrong_guesses)
                display_answer(answer)
                print(Fore.RED + "💀 You lost!💀")
                is_running = False
        
        play_again = input("Want to play again? (y/n): ").lower()
        if play_again != "y":
            print("Thank you for playing! Play again soon!")
            break  # Exit the game loop if they don't want to play again

if __name__ == "__main__":
    main()
